### ISSUE TITLE
no-extra-parens error when doing a type assertion

### ISSUE THREAD

**Issue by JamesHenry** [typescript-eslint/typescript-eslint#12]
This issue was initially reported here: https://github.com/eslint/typescript-eslint-parser/issues/465

---

**What version of TypeScript are you using?**

2.8.1

**What version of `typescript-eslint-parser` are you using?**

14.0.0

**What code were you trying to parse?**

```ts
t.true((me.get as SinonStub).calledWithExactly('/foo', other));
```

**What did you expect to happen?**

ESLint should not complain.

**What happened?**

`error |   Gratuitous parentheses around expression     |        no-extra-parens`

**Comment by iapYang** [comment]
Got eslint error:
Definition for rule '@typescript-eslint/no-extra-parens' was not found

What version of TypeScript are you using?

3.4.3

What version of typescript-eslint-parser are you using?

1.6.0
What version of typescript-eslint-plugin are you using?

1.6.0
